### PR TITLE
fix(eks): Could not use ec2 instance type and size that their names contains dashes

### LINKED ISSUE #27587: aws-cdk-lib/aws-ec2: "Malformed instance type identifier" when creating ec2 instance of type 'm7i-flex.large'
### Describe the bug

I'm trying to add an `m7i-flex.large` instance to a managed node group per
`createManagedComputeNodes(
        name,
        subnet,
        [
          new ec2.InstanceType('m7i-flex.large'),
        ],
        eks.CapacityType.SPOT,
      );`

But receive a `Malformed instance type identifier` when doing so.

### Expected Behavior

I'm able to add the `m7i-flex.large` instance type as an instance choice for the managed node group

### Current Behavior

```
Error: Malformed instance type identifier
    at InstanceType.sameInstanceClassAs (/node_modules/aws-cdk-lib/aws-ec2/lib/instance-types.js:1:17751)
    at /node_modules/aws-cdk-lib/aws-eks/lib/managed-nodegroup.js:1:7443
    at Array.some (<anonymous>)
    at isGpuInstanceType (/node_modules/aws-cdk-lib/aws-eks/lib/managed-nodegroup.js:1:7422)
    at typeToArch (/node_modules/aws-cdk-lib/aws-eks/lib/managed-nodegroup.js:1:8182)
    at Array.map (<anonymous>)
    at getPossibleAmiTypes (/node_modules/aws-cdk-lib/aws-eks/lib/managed-nodegroup.js:1:8442)
    at new Nodegroup (/node_modules/aws-cdk-lib/aws-eks/lib/managed-nodegroup.js:1:3543)
    at Cluster.addNodegroupCapacity (/node_modules/aws-cdk-lib/aws-eks/lib/cluster.js:1:19557)

Node.js v20.5.0
```

### Reproduction Steps

Add a `m7i-flex.large` instance type to a managed node group per "Describe the bug".  Then run `cdk synthesize`.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.101.1 (build 16ddad1)

### Framework Version

_No response_

### Node.js Version

20.5.0

### OS

MAC OS Monterey 12.6.1

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by moelasmar**
Fix the Ec2 instance class regular expression to accept values that contains dashes like (m7i-flex, and metal-48xl).

Closes #27587.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster.ts [hunk 1]
```diff
@@ -223,7 +223,7 @@ class EksClusterStack extends Stack {
       instanceTypes: [
         new ec2.InstanceType('c5.large'),
         new ec2.InstanceType('c5a.large'),
-        new ec2.InstanceType('c5d.large'),
+        new ec2.InstanceType('m7i-flex.large'),
```
Comment (vinayak-kukreja) [inline packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster.ts:226]: Were you able to run this integ test and deploy the resource to your account?

**Review by vinayak-kukreja — CHANGES_REQUESTED** [review]
Hey, thank you for opening a contribution with us. 

Couple of comments,
* The PR title, when it is a fix should tell more about the error/issue and not the solution. So for instance, `fix(ec2): instance type and size regex does not accommodate dashes`
* Could you add to integ test present in ec2 to verify we are able to provision these instances. If you are unable to run these test on your account, please let me know and I can verify those.

File: packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster.ts (hunk 1 above)
Comment (moelasmar) [reply packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster.ts:226]: I was able to run the test case, but I did not try to deploy it.

**Comment by moelasmar** [comment]
> Hey, @moelasmar thank you for opening a contribution with us.
> 
> Couple of comments,
> 
> * The PR title, when it is a fix should tell more about the error/issue and not the solution. So for instance, `fix(ec2): instance type and size regex does not accommodate dashes`
> * Could you add to integ test present in ec2 to verify we are able to provision these instances. If you are unable to run these test on your account, please let me know and I can verify those.

The problem is mainly related to adding new managed node groups to an EKS Cluster, and not while creating EC2 instances. The problem is mainly in these 2 functions:
https://github.com/aws/aws-cdk/blob/3b3fc3aa545b18cf0c54715426d00fe909c44cb5/packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts#L1517-L1544

Where are mainly used in these functions in eks managed nodegroup

https://github.com/aws/aws-cdk/blob/f9ee5e651a461fa7995823b556aa3b7c69c59a72/packages/aws-cdk-lib/aws-eks/lib/managed-nodegroup.ts#L559-L579

Anyway, I will try to add a new Ec2 test case

File: packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster.ts (hunk 1 above)
Comment (vinayak-kukreja) [reply packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster.ts:226]: Thank you. Let me know if I can help.

**Comment by vinayak-kukreja** [comment]
Hey @moelasmar , I ran the integ test and have added that. Looks like the `fork` needs to be updated. Could you please do that and update here? I can resolve the other conflicts.

**Comment by moelasmar** [comment]
> Hey @moelasmar , I ran the integ test and have added that. Looks like the `fork` needs to be updated. Could you please do that and update here? I can resolve the other conflicts.

I found some issues on resolving the conflicts, so I start from a fresh fork of the main branch, and added my changes again.
I also executed `yarn integ --directory test/aws-eks/test --update-on-failed` to run the integration test cases and update the snapshot, and also executed ` yarn test aws-ec2` to verify the unit test cases.

**Review by paulhcsun — APPROVED** [review]

**Comment by paulhcsun** [comment]
@Mergifyio update

**Pull request merged**